api/service.go

```diff
@@ -785,88 +834,76 @@ func (s *service) AddFile(server pb.API_AddFileServer) error {
 		return err
 	}
 
+	buckPath := path.New(buck.Path)
 	dirpth, err := s.ipfsClient.Object().
-		AddLink(server.Context(), folderPath, fileName, pth, options.Object.Create(true))
+		AddLink(server.Context(), buckPath, fileName, pth, options.Object.Create(true))
 	if err != nil {
 		return err
 	}
-	if err = s.ipfsClient.Pin().Update(server.Context(), folderPath, dirpth); err != nil {
+	if err = s.ipfsClient.Pin().Update(server.Context(), buckPath, dirpth); err != nil {
 		return err
 	}
-	if err = s.collections.Folders.UpdatePath(server.Context(), folder, dirpth); err != nil {
+
+	buck.Path = dirpth.String()
+	buck.Updated = time.Now().Unix()
+	if err = s.collections.Buckets.Save(server.Context(), buck); err != nil {
 		return err
 	}
 
-	if err = sendEvent(&pb.AddFileReply_Event{
+	if err = sendEvent(&pb.PushBucketPathReply_Event{
 		Path: pth.String(),
 		Size: size,
+		Root: &pb.BucketRoot{
+			Name:    buck.Name,
+			Path:    buck.Path,
+			Created: buck.Created,
+			Updated: buck.Updated,
+			Public:  false,
+		},
 	}); err != nil {
 		return err
 	}
 
-	log.Debugf("added file %s to folder with new path: %s", pth.String(), dirpth.String())
+	log.Debugf("pushed %s to bucket: %s", fileName, buck.Name)
 	return nil
 }
```

Comment:
Q: `PushBucketPath` should be understood as an overriding operation. What it was asked, will be done.

Reply:
Yep. As a future thing, maybe we can enable a "force" type flag that is required to overwrite paths.

Reply:
Yeah, overriding policies is another big story. 

Brainstorm: An intermediate idea before some fully-mergable magic algorithm: send the last known bucket root while pushing... the server can check is overriding from that rootcid. If that isn't the case, fail since will override some not-known other change. (Similar of a fast-forward check, if not -> fail). Sounds that might cover most of the bad-cases and would be quite easy.

Reply:
Nice, I like it. I'll create a ticket.